src/models/tests/m_identity_tests.erl

```diff
@@ -0,0 +1,107 @@
+%% @author Maas-Maarten Zeeman <mmzeeman@xs4all.nl>
+%% @hidden
+
+-module(m_identity_tests).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("zotonic.hrl").
+
+hash_test() ->
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash("1234")),
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash(<<"1234">>)),
+    ?assertMatch({bcrypt, <<"$2a$12$", _/binary>>}, m_identity:hash(<<195, 154, 105, 105>>)),
+    ok.
+
+needs_rehash_test() ->
+    ?assertEqual(true, m_identity:needs_rehash(old_hash("password"))),
+    ?assertEqual(false, m_identity:needs_rehash(m_identity:hash("password"))),
+    ok.
+
+hash_is_equal_test() ->
+    PasswordHash = m_identity:hash("password"),
+
+    ?assert(m_identity:hash_is_equal("password", PasswordHash)),
+    ?assert(not m_identity:hash_is_equal("1234", PasswordHash)),
+
+    ok.
+
+hash_is_equal_old_hash_test() ->
+    PasswordHash = old_hash("password"),
+
+    % Validating stored old password hashes should still work.
+    ?assert(m_identity:hash_is_equal("password", PasswordHash)),
+    ?assert(not m_identity:hash_is_equal("1234", PasswordHash)),
+
+    ok.
+
+check_password_no_user() ->
+    C = z_context:new(testsandboxdb),
+    AdminC = z_acl:logon(?ACL_ADMIN_USER_ID, C),
+    ok = delete_user("mr_z", AdminC),
+
+    ?assertEqual({error, nouser}, m_identity:check_username_pw("mr_z", "does-not-exist", C)),
+    ok.
+
+check_username_password_test() ->
+    C = z_context:new(testsandboxdb),
+    AdminC = z_acl:logon(?ACL_ADMIN_USER_ID, C),
+
+    MrYId = ensure_new_user("mr_y", "secret", AdminC),
+
+    ?assertEqual({ok, MrYId}, m_identity:check_username_pw("mr_y", "secret", C)),
```

Comment:
Strange, the tests run over here. 

It looks like out of data output because of the module dependency warnings. I don't get those when I run the tests locally.